pkg/provider/data_sync_test.go

```diff
@@ -1,29 +1,94 @@
+/*
+Copyright 2020 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package provider
 
 import (
+	"fmt"
+	"regexp"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-func TestAccSync_basic(t *testing.T) {
+func TestAccDataSync_basic(t *testing.T) {
+	resourceName := "data.flux_sync.main"
 	resource.ParallelTest(t, resource.TestCase{
-		Providers: testAccProviders,
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccSyncBasicDataSource,
+				// Without required target_path set
+				Config:      `data "flux_sync" "main" {  url = "ssh://git@example.com" }`,
```

Comment:
I think we should break these one line strings into a separate constants if they contain variables. It makes them easier to edit later on.